### DISCUSSION TITLE
Macro to easily create file in cwd.

### DISCUSSION THREAD

**Discussion by EricHenry** [helix-editor/helix#12002] [Show and tell]
This is a super handy mapping for creating a file in your current working directory. Uses a combination of the keybinding to [select a register](https://docs.helix-editor.com/master/keymap.html#insert-mode) and [prompt key-bindings](https://docs.helix-editor.com/master/keymap.html#prompt) to fill in the prompt with the current path minus the current file.

```
[keys.normal.space]
o = "@:o <C-r>%<C-w>"
```

![image](https://github.com/user-attachments/assets/3e140bd3-b686-4ec8-8971-8611477652dd)

**Comment by David-Else** [comment]
On master I get the following, with nothing for the home dir `~`, I have also merged https://github.com/helix-editor/helix/pull/2608 which maybe to blame:

![image](https://github.com/user-attachments/assets/1df6e0bd-4b3f-4a7a-a5bb-a668631ebfec)

Is there a bug now?

**Reply by EricHenry** [reply]
hmmmm maybe? I am just running on master without any other branches merged in.

which directory did you open up helix in?

**Reply by David-Else** [reply]
`~/.config/helix/`, just opened my Helix config in the config dir.

EDIT I added a comment in the PR https://github.com/helix-editor/helix/pull/2608#issuecomment-2457780573

**Reply by EricHenry** [reply]
Thanks for doing that! are you able to test on just the master branch, to verify?

**Reply by David-Else** [reply]
> Thanks for doing that! are you able to test on just the master branch, to verify?

I am a bit busy, I am sure if it works on your master and not mine with the path completion PR then it must be the PR. I am using Linux.

**Reply by EricHenry** [reply]
yeah same here, also on linux.

**Reply by gabydd** [reply]
Hmm which directories is your current working directory the .config/helix without ~/ should be fine if your helix cwd is ~

**Reply by David-Else** [reply]
I don't remember how I originally opened the file, but If I open `config.toml` from the command line inside the config dir:

![image](https://github.com/user-attachments/assets/e61f594c-88e3-4862-a758-d53dcbad89cc)

Run the macro and get:

![image](https://github.com/user-attachments/assets/8b2cdb2d-18c6-44d4-9b46-6d28ae079809)

I can now enter a filename and save it, and it does appear in the CWD.

I am not sure why @EricHenry is getting the `~`, I am just opening from Kitty inside the directory itself.

**Reply by EricHenry** [reply]
I didn't open helix in the `~` directory. I was in a project directory, then opened the config file, lastly I triggered the macro. That may be why it shows the path with `~`

**Comment by nik-rev** [comment]
I've made a PR which adds a register `/` specifically for this purpose, which contains the parent directory of the current file. 

It'll make creating files a lot more nice if it's merged. Especially since on master the `%` register doesn't seem to provide the absolute path, but just the name of the current file.

https://github.com/helix-editor/helix/pull/12037
